src/core/qgsziputils.cpp

```diff
@@ -84,8 +85,19 @@ bool QgsZipUtils::unzip( const QString &zipFilename, const QString &dir, QString
         char *buf = new char[len];
         if ( zip_fread( file, buf, len ) != -1 )
         {
-          QFileInfo newFile( QDir( dir ), QString( stat.name ) );
-          std::ofstream( newFile.absoluteFilePath().toStdString() ).write( buf, len );
+          QString fileName( stat.name );
+          QFileInfo newFile( QDir( dir ), fileName );
+          bool isDirectory = fileName.endsWith( '/' );
+
+          if ( isDirectory )
+          {
+            if ( !QDir( dir ).mkdir( fileName ) )
+              QgsDebugMsg( QString( "Failed to create a subdirectory %1/%2" ).arg( dir ).arg( fileName ) );
```

Comment:
QStringLiteral( "Failed to create a subdirectory %1/%2" ).arg( dir, fileName )

Reply:
Actually, should that go to the log rather than a debug message (which is not available on release builds)?

Reply:
I am easy, it can be promoted to log messages.

Reply:
I think it would be good.
Unless there is a good reason to assume that this never happens in reality.